### PR TITLE
Add automatic documentation deployment workflow

### LINKED ISSUE #385: Automatic documentation deployment
# Automatic documentation deployment

Currently, updating the [Tangelo documentation](http://tangelo-docs.goodchemistry.com/) involves significant manual work. We would like to have an automatic deployment workflow, similar to what we have set up in the [Tangelo-Examples](https://github.com/goodchemistryco/Tangelo-Examples) repo.

**What problem does this feature request help you overcome?**
As of now, regenerating or updating the documentation requires performing the steps outlined in [dev_tools/build_sphinx_docs.sh](https://github.com/goodchemistryco/Tangelo/blob/main/dev_tools/build_sphinx_docs.sh) locally, and then uploading the files to an AWS bucket. It's cumbersome and manual, and sometimes it may not be updated right away.

**Describe the solution you'd like**
We would like an automatic workflow (similar to the one in the Tangelo-Examples repo, see files below) to compile and upload the documentation using GitHub Pages every time the `main` branch is updated, to guarantee that our documentation is always reliable !

- The documentation left behind by past versions of ourselves should be helpful here: https://github.com/goodchemistryco/Tangelo-Examples/blob/main/dev_tools/generate_documentation.rst
- An "automated workflow" file we should add to the repo is available [here](https://github.com/goodchemistryco/Tangelo-Examples/blob/main/.github/workflows/publish.yml)
- There are also other [online resources](https://coderefinery.github.io/documentation/gh_workflow/) that may be relevant.

The source files would reside in a `gh-pages` branch, and the website could be accessed via a GitHub Pages address `https://goodchemistryco.github.io/Tangelo`. After that, we would only need to point `http://tangelo-docs.goodchemistry.com/` to the corresponding GitHub page. (this address will be changed to a SandboxAQ address later, don't worry about it 😄 )

### PR DESCRIPTION
**Pull request by king-p3nguin**
Resolve #385 

The automatically deployed documentation is available at: https://king-p3nguin.github.io/Tangelo/

Settings for GitHub Pages is as follows:
image

(I am contributing to this project as a unitaryHACK participant)

### CODE REVIEW

File: .github/workflows/deploy_docs.yml
Comment (ValentinS4t1qbit) [inline .github/workflows/deploy_docs.yml:4]: Would it be possible to also add a manual trigger for that workflow, by any chance ?

File: .gitignore
Comment (alexfleury-sb) [inline .gitignore:1]: Is this file necessary for ignoring temp file when generating the documentation? Or is it for your convenience to not commit those?

@ValentinS4t1qbit we never defined a `gitignore` file yet, it could be the occasion to merge one.

Comment (ValentinS4t1qbit) [reply .gitignore:1]: @alexfleury-sb I believe pycache folders contain the compiled Python code (compiled to C or something, pyc files).  They should never be committed in this repo, it's stuff thats generated at runtime by Python when you run anything. I'm very fine with this change

File: .github/workflows/deploy_docs.yml
Comment (king-p3nguin) [reply .github/workflows/deploy_docs.yml:4]: I have added a `workflow_dispatch` trigger, so authorized people should be able to run the workflow manually as well.

A manual trigger should appear on actions tab like this:
image

**Comment by king-p3nguin** [comment]
@ValentinS4t1qbit When I tried running this workflow on my fork, `peaceiris/actions-gh-pages@v4` automatically created `gh-pages` branch, so I think maintainers do not need to create this branch manually.

**Comment by ValentinS4t1qbit** [comment]
> @ValentinS4t1qbit When I tried running this workflow on my fork, `peaceiris/actions-gh-pages@v4` automatically created `gh-pages` branch, so I think maintainers do not need to create this branch manually.

@king-p3nguin I cannot select it like in your screenshot, the drop-down menu only offer existing branch. Maybe a better question is: why wouldn't we select `main` ? (I do not quite recall why the issue description stated `gh-pages`, maybe we did not quite get how to do this right).

**Comment by king-p3nguin** [comment]
@ValentinS4t1qbit
This workflow is not executed until the yaml file is pushed to the main branch, so the `gh-pages` branch has not been created yet. It should be created through the workflow as soon as this PR is merged.

I think the reason `peaceiris/actions-gh-pages@v4` pushes documentation to the separate `gh-pages` branch instead of the main branch is to simplify the commit history.

**Comment by ValentinS4t1qbit** [comment]
@king-p3nguin Alright, I'll give it a shot. The finish line for the hack is closing the issue, it is ok it we somehow need two separate PRs for adjustments.

**Review by ValentinS4t1qbit — APPROVED** [review]

**Pull request merged**
